.github/workflows/ci-workflow.yml

```diff
@@ -66,7 +66,7 @@ jobs:
       run: pip install dataclasses
 
     - name: Install pytest
-      run: pip install pytest~=6.2.5
+      run: pip install pytest
```

Comment:
I assume this will install the latest version. Usually, I recommend pinning things in CI to prevent things from suddenly breaking. I assume you changed this because you want things to work with the latest version?

Reply:
Yes.  Sadly, the latest version is the one that can use `tomli`.  Earlier versions use the `toml` lib instead. However, I could pin it to the latest version (7.1).

I did have it pinned to 6.2.5 for the very reason you cited - it upgraded itself a while ago and broke everything.  

However, due to a nasty combination of https://github.com/psf/black/issues/2964 for Black (the formatter used in the `generate_tests.py`) and the move toward `tomli` for toml parsing in the community, I sorta had to upgrade.  

Both the latest version of Black (_needed in order to fix the _unicodefun error_) and the latest version of Pytest use `tomli`.  Once I changed `data.py` to use `tomli` for Black, I needed to then upgrade pytest as well..

Reply:
> However, I could pin it to the latest version (7.1).

That could be a good idea. I'll let you decide of course :)